parsers/powershell.c

```diff
@@ -619,12 +624,20 @@ static void enterScope (tokenInfo *const parentToken,
 				enterScope (token, NULL, KIND_GHOST_INDEX);
 				break;
 
-			case TOKEN_KEYWORD_FUNCTION:
-				readNext = parseFunction (token);
-				break;
+			case TOKEN_KEYWORD:
+				switch (token->keyword)
+				{
+					case KEYWORD_function:
+					case KEYWORD_filter:
```

Comment:
It seems that there is no difference between filter and function FOR CTAGS though they have their own purpose in PowerShell.

In that case, you can map "function" and "filter" to KEYWORD_function; and we can make KEYWORD_filter obsoleted. In the other words, let's ctags extract a filter object with function kind. Let's call this "unifying-function-and-filter approach".

If you think the difference between function and filter is important for ctags' users,
introducing filter kind is another approach.
Let's call "introducing-a-kind-for-filter approach".

"unifying-function-and-filter approach" is easy, and we can choose "introducing-a-kind-for-filter approach" later. So, I'll make more comments about "unifying-function-and-filter approach" in other places.

Reply:
The short answer is that I think the "introducing-a-kind-for-filter approach" should be chosen.

If a user writes function and filter separately, there is a reason. Therefore, function and filter should be treated differently. 

There are reasons why we should not unify function and filter.

In the Before You Start section of the Contribution Guide, it says that ctag users use ctag for `navigating the source code tree`.
https://docs.ctags.io/en/latest/contributions.html#before-you-start

At least in PowerShell v5.1, function and filter use the same namespace. Thus, if `function func1 {` and `filter func1 {` exist, the later definition is valid.

If there are multiple functions with the same name in the source code, some implementations of the `source code tree` may display strange results. However, if they are of the same kind, the user will easily notice it because func1 will be displayed near func1 in the code tree.

Similarly, the `source code tree` may look strange when there is a function and a filter with the same name in the source code. I would guess that many implementations will have the names appear farther apart, especially when the kind is different. Because of the distance, I would think that it would be difficult to notice that the names are identical.

I think the mechanism for noticing duplicate function and filter names is more the role of a separate linter than the role of ctags.

As a result, I think the "introducing-a-kind-for-filter approach" should be chosen.

Reply:
I agree with this comment.
https://github.com/universal-ctags/ctags/pull/3480#issuecomment-1246010526

Reply:
So will you update the pull request that follows the "introducing-a-kind-for-filter approach" ?

Reply:
Thank you for your kind review and comments.
I have implemented it and git pushed it.
Please check it out.

Please let me know if anything is missing.